server/v2/core/appmanager/app.go

```diff
@@ -7,9 +7,14 @@ import (
 	"google.golang.org/protobuf/proto"
 
 	"cosmossdk.io/server/v2/core/event"
+	"cosmossdk.io/server/v2/core/store"
 	"cosmossdk.io/server/v2/core/transaction"
 )
 
+// PrepareHandler is a function that takes a list of transactions and returns a list of transactions and a list of changesets
+// In the block building phase the transactions from the mempool can be verified, executed right away or lazily.
+type PrepareHandler[T transaction.Tx] func(context.Context, []T, store.ReadonlyState) ([]T, []store.ChangeSet, error)
```

Comment:
So I think I asked this in the other PR (I can't seem to find it). What is supposed to happen with the `[]ChangeSet`?

Also, consider making it `ChangeSet` and not `[]ChangeSet`, as the former indicates it's already a set, the latter indicates it's a set of a set.

Reply:
discussed in a call, we will remove the notion of executin in prepare, if a team wants a different execution they would do it through consensus